config/runtime/server.go

```diff
@@ -80,83 +82,59 @@ func BuildEntrypointHandlers(conf *config.Gateway, httpConf *HTTPConfig, log *lo
 
 	accessControls := configureAccessControls(conf)
 
-	handlers := make(EntrypointHandlers, 0)
+	server := make(Server, 0)
 
-	var err error
-	for idx, server := range conf.Server {
-		configureBasePathes(server)
-
-		mux := &Mux{
-			APIErrTpl: errors.DefaultJSON,
-			FSErrTpl:  errors.DefaultHTML,
+	for _, srvConf := range conf.Server {
+		muxOptions, err := NewMuxOptions(srvConf)
+		if err != nil {
+			log.Fatal(err)
 		}
 
-		if server.Files != nil {
-			if server.Files.ErrorFile != "" {
-				if mux.FSErrTpl, err = errors.NewTemplateFromFile(getAbsPath(server.Files.ErrorFile, log)); err != nil {
-					log.Fatal(err)
+		var spaHandler http.Handler
+		if srvConf.Spa != nil {
+			spaHandler, err = handler.NewSpa(srvConf.Spa.BootstrapFile)
+			if err != nil {
+				log.Fatal(err)
+			}
+			for _, spaPath := range srvConf.Spa.Paths {
+				for _, p := range getPathsFromHosts(defaultPort, srvConf.Hosts,
+					utils.JoinPath(srvConf.BasePath, srvConf.Spa.BasePath, spaPath)) {
```

Comment:
Because of the hosts path pattern. Same in api endpoint pattern, without leading `/`. See `getPathsFromHosts`.

The real issue here is `files` has no host specific routing. I will fix that.

Reply:
`getPathsFromHosts()`:

```
if len(list) == 0 {
	list = []string{path}
}
```

The path can be an empty string `""` or a string _without_ leading slash `/`.